hpc/job.py

```diff
@@ -101,13 +112,15 @@ def __init__(
         HBP_APP_ID: Optional[str] = None,
         HBP_APP_SECRET: Optional[str] = None,
         HPC_PATH_PREFIX: Optional[str] = None,
+        HPC_PROJECT_NAME: Optional[str] = None,
         access_token: Optional[str] = None,
     ):
         self.access_token = access_token
         self.HBP_REFRESH_TOKEN = HBP_REFRESH_TOKEN or os.environ["HBP_REFRESH_TOKEN"]
         self.HBP_APP_ID = HBP_APP_ID or os.environ["HBP_APP_ID"]
         self.HBP_APP_SECRET = HBP_APP_SECRET or os.environ["HBP_APP_SECRET"]
         self.HPC_PATH_PREFIX = HPC_PATH_PREFIX or os.environ.get("HPC_PATH_PREFIX", "")
+        self.HPC_PROJECT_NAME = HPC_PROJECT_NAME or os.environ.get("HPC_PROJECT_NAME", None)
```

Comment:
just wondering about the default values here... Technically this allows instantiating a totally invalid `HpcEnvironment`?

Reply:
I think I was a little misguided when I wrote a lot of this code; I don't think the os environment should take such a prominent part of the code anymore, as that is more of a user interface thing.... but I guess I was thinking too much in openstack-cli mode -.-

Anyways, if you instantiate an `HpcEnvironment`  with no parameters you'll get a runtime error if you don't have the correct variables set in your environment. It's definitely not as good as static validation, but at least the code will fail immediately. For the values that actually accept a `None`, like the new `HPC_PROJECT_NAME`, the idea is that we can use this same code to run in different HPCs, and some of them might not need a project name (or not even allow it)

Reply:
If we are about to refactor this, I would suggest using something like [Pydantic](https://pydantic-docs.helpmanual.io/) for the data model, and maybe [dotenv](https://pypi.org/project/python-dotenv/) for specifying those values through environment variables.

Reply:
thx for the explanation @Tomaz-Vieira (an sorry for not checking the code following that holds the validation logic).
@emilmelnikov, cool, didn't know Pydantic, and dynaconf.

Reply:
I'll have a look at all of Pydantic, dotenv and friends =)
But for now, we should get this going so that we can fire jobs on CSCS again